runtime/common/MeasureCounts.cpp

```diff
@@ -127,31 +139,19 @@ std::vector<std::size_t> sample_result::serialize() const {
 
 void sample_result::deserialize(std::vector<std::size_t> &data) {
   std::size_t stride = 0;
+  totalShots = 0;
+
   while (stride < data.size()) {
-    auto nChars = data[stride];
-    stride++;
-    std::string name = "";
-    for (std::size_t i = 0; i < nChars; i++)
-      name += std::string(1, char(data[stride + i]));
-
-    stride += nChars;
-    std::size_t localShots = 0;
-    std::unordered_map<std::string, std::size_t> localCounts;
-    auto nBs = data[stride];
-    stride++;
-    for (std::size_t j = stride; j < stride + nBs * 3; j += 3) {
-      auto bitstring_as_long = data[j];
-      auto size_of_bitstring = data[j + 1];
-      auto count = data[j + 2];
-      auto bs = longToBitString(size_of_bitstring, bitstring_as_long);
-      localShots += count;
-      localCounts.insert({bs, count});
-    }
+    std::string name = extractNameFromData(data, stride);
 
-    sampleResults.insert({name, ExecutionResult{localCounts, name}});
+    std::unordered_map<std::string, std::size_t> localCounts;
+    deserializeCounts(data, stride, localCounts);
 
-    stride += nBs * 3;
-    totalShots = localShots;
+    sampleResults.insert({name, ExecutionResult{std::move(localCounts), name}});
+    totalShots += std::accumulate(
+        localCounts.begin(), localCounts.end(), 0,
+        [](std::size_t sum, const auto &pair) { return sum + pair.second; });
+    stride += localCounts.size() * 3;
```

Comment:
It is unsafe to use `localCounts` after invoking `std::move()` on it because it is undefined whether or not it will retain its values after being "moved".